src/Components/CommentAdder.jsx

```diff
@@ -0,0 +1,45 @@
+import { postComment } from "../../utils";
+import { useState } from "react";
+import { useParams } from "react-router";
+
+function CommentAdder({setArticleComments, currentUser, setFetchedComments}) {
+
+    const [isLoading, setIsLoading] = useState(true)
+const {username} = currentUser
+const [newComment, setNewComment] = useState("")
+const [commentMessage, setCommentMessage] = useState("")
+const {article_id }= useParams()
+ 
+function handleSubmit(event) {
+    setFetchedComments(false)
+    event.preventDefault()
+    postComment(username, newComment, article_id).then((newCommentFromApi) => {
+        setNewComment("")
+        setFetchedComments(true)
+        setCommentMessage("Comment posted successfully.")
+        setTimeout(() => {
+            setCommentMessage("")
+        }, 3000)
+         setIsLoading(false)
+        setArticleComments((currComments) => {
+            return [newCommentFromApi, ...currComments]
+        })
```

Comment:
Please include a .catch() block and let the user know when a post request fails.